### DISCUSSION TITLE
Possible to make a plugin that behaves like the veinminer tmod?

### DISCUSSION THREAD

**Discussion by MSPaintRose** [Pryaxis/TShock#2009] [Q&A]
title

**Comment by zaneshaw** [comment]
I don't think that's possible from what I know.

**Comment by moisterrific** [comment]
What’s veinminer?

**Reply by Kojirremer** [reply]
I haven't used it myself, so I can't be sure if I got it right, but what I remember about its description is that when you found an ore vein, you just had to mine a single tile for the whole vein to drop in its item form.

**Comment by moisterrific** [comment]
Oh so kinda like this?

https://github.com/moisterrific/RecursivePick

It’s outdated but I give it a try at updating it.

**Reply by Kojirremer** [reply]
Yes, I think that's exactly what it's supposed to do. I would remove the ability to do the same with walls, though. It would make destroying certain wall-depending biomes like the dungeon, spider caves and the underground desert way too easy.

**Reply by MSPaintRose** [reply]
Oh wow thanks for finding this. Just need to learn how to use the Terraria API to upgrade this.


Thanks

**Comment by YSpoof** [comment]
Any wants to upgrade that for TShock 5? Idk C# :(

https://github.com/Megghy/VeinMiner

**Comment by YSpoof** [comment]
Yup, I got it updated and fully working on TShock 5 :)

I've added the DLL to the releases on my repo util Megghy adds it to their own:

https://github.com/YSpoof/TSHockVeinMiner
